src/zenml/service_connectors/service_connector_utils.py

```diff
@@ -0,0 +1,346 @@
+#  Copyright (c) ZenML GmbH 2024. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Utility methods for Service Connectors."""
+
+from typing import Dict, List, Union
+from uuid import UUID
+
+from zenml.client import Client
+from zenml.enums import StackComponentType
+from zenml.models.v2.core.service_connector import ServiceConnectorRequest
+from zenml.models.v2.misc.full_stack import (
+    ResourcesInfo,
+    ServiceConnectorInfo,
+    ServiceConnectorResourcesInfo,
+)
+from zenml.utils.pagination_utils import depaginate
+
+
+def _prepare_resource_info(
+    connector_details: Union[UUID, ServiceConnectorInfo],
+    resource_ids: List[str],
+    stack_component_type: StackComponentType,
+    flavor: str,
+    required_configuration: Dict[str, str],
+    flavor_display_name: str,
+    use_resource_value_as_fixed_config: bool = False,
+) -> ResourcesInfo:
+    existing_components = []
+    if isinstance(connector_details, UUID):
+        existing_components = depaginate(
+            Client().list_stack_components,
```

Comment:
Should we include the flavor here as well? RIght now, this would give me all orchestrators for a service connector, even though the resource info is scoped to a specific flavor, correct?

Reply:
That's right, good catch 👍🏼